src/Silex/ControllerCollection.php

```diff
@@ -57,11 +59,21 @@ public function __construct(Route $defaultRoute, $routesFactory = null)
     /**
      * Mounts controllers under the given route prefix.
      *
-     * @param string               $prefix      The route prefix
-     * @param ControllerCollection $controllers A ControllerCollection instance
+     * @param string                        $prefix      The route prefix
+     * @param ControllerCollection|callable $controllers A ControllerCollection instance or a callable for defining routes
+     *
+     * @throws \LogicException
      */
-    public function mount($prefix, ControllerCollection $controllers)
+    public function mount($prefix, $controllers)
     {
+        if (is_callable($controllers)) {
+            $collection = $this->controllersFactory ? call_user_func($this->controllersFactory) : new static(new Route(), new RouteCollection());
+            call_user_func($controllers, $collection);
+            $controllers = $collection;
```

Comment:
Personally, I like the syntax of the first option, but the silex src code is a little less maintainable :(. I'm ok with the second option. Definitely is a much cleaner patch into the codebase; however, we'd have to clearly let users know that the mount with a function **only** works from the `Silex\Application`